### PR TITLE
[INTCONTRIB-37] Module rename.

### PR DESCRIPTION
**Pull request by govindmaloo**
https://salsadigital.atlassian.net/browse/INTCONTRIB-37

### CODE REVIEW

**Comment by govindmaloo** [comment]
@sonnykt CI Tests are not configured properly, so failed. Let me know once fixed I will review them again.

File: js/alert_banner.js [hunk 1]
```diff
@@ -42,10 +42,10 @@
     return true;
   }
 
-  Drupal.behaviors.govcmsAlertsRestBlock = {
+  Drupal.behaviors.AlertsRestBlock = {
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
  Drupal.behaviors.AlertBannersRestBlock = {
```

File: js/alert_banner.js [hunk 2]
```diff
@@ -149,7 +149,7 @@
 
               $alert.appendTo($placeholder);
             }
-            Drupal.behaviors.govcmsAlertsRestBlock.attach(context, settings);
+            Drupal.behaviors.AlertsRestBlock.attach(context, settings);
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
            Drupal.behaviors.AlertBannersRestBlock.attach(context, settings);
```

File: js/alert_banner.js [hunk 3]
```diff
@@ -61,17 +61,17 @@
...
-      $('.govcms-alerts article.node--type-alert button.govcms-alert-close', context).click(function (event) {
+      $('.alert-banners article.node--type-alert button.alert-close', context).click(function (event) {
         var alert_id = $(event.target).attr('data-alert-id');
         $.cookie('hide_alert_id_' + alert_id, true);
         $('article.node--type-alert[data-alert-id="' + alert_id + '"]').remove();
       });
 
       // Loads the alerts for REST endpoint.
-      $('.govcms-alerts:not(.alerts-processed)', context).once('govcms_alerts_load').each(function (index, element) {
+      $('.alert-banners:not(.alerts-processed)', context).once('alerts_load').each(function (index, element) {
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
      $('.alert-banners:not(.alerts-processed)', context).once('alert_banners_load').each(function (index, element) {
```

File: js/alert_banner.js [hunk 4]
```diff
@@ -61,17 +61,17 @@
...
         var endpoint = $(element).attr('data-alert-endpoint');
         if ((typeof endpoint == 'undefined') || !endpoint || endpoint.length === 0) {
-          endpoint = '/govcms-alerts?_format=json';
+          endpoint = '/alerts?_format=json';
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
          endpoint = '/alert-banners?_format=json';
```

File: alert_banner.module [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  // Do not show alerts on admin pages.
+  if (\Drupal::service('router.admin_context')->isAdminRoute() &&
+    !\Drupal::config('alert_banner.settings')->get('show_on_admin')) {
+    return;
+  }
+
+  $entity_type_manager = \Drupal::entityTypeManager();
+
+  // Retrieve the endpoint of GovCMS Alert Rest.
+  $endpoint = '/alerts?_format=json';
```
Comment (sonnykt) [inline alert_banner.module]:
```suggestion
  $endpoint = '/alert-banners?_format=json';
```

File: alert_banner.module [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+  // Do not show alerts on admin pages.
+  if (\Drupal::service('router.admin_context')->isAdminRoute() &&
+    !\Drupal::config('alert_banner.settings')->get('show_on_admin')) {
+    return;
+  }
+
+  $entity_type_manager = \Drupal::entityTypeManager();
+
+  // Retrieve the endpoint of GovCMS Alert Rest.
```
Comment (sonnykt) [inline alert_banner.module]: @govindmaloo No mention of GovCMS pls.

File: alert_banner.routing.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+alert_banner.alert_banner_config_form:
+  path: '/admin/config/system/alert_banner'
+  defaults:
+    _form: '\Drupal\alert_banner\Form\ConfigForm'
+    _title: 'Alert Settings'
```
Comment (sonnykt) [inline alert_banner.routing.yml]:
```suggestion
    _title: 'Alert Banner Settings'
```

File: README.md [hunk 8]
```diff
@@ -43,12 +43,12 @@ The GovCMS Alert module provides the site-wide alert messages and announcements.
       * Permanent, then
       * Node Sticky, then
       * Alert date (newest first).
```
Comment (sonnykt) [inline README.md]: @govindmaloo Alert date no longer exists, and the `7 days` logic should also be removed from readme.

All references to the Date field must be removed.

File: config/install/core.date_format.alert_date.yml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 langcode: en
 status: true
 dependencies: {  }
-id: govcms_alert_date
+id: alert_date
```
Comment (sonnykt) [inline config/install/core.date_format.alert_date.yml]:
```suggestion
id: alert_banner_date
```

All references of `govcms_alert` must be changed to `alert_banner`, not `alert`.

File: config/install/core.entity_view_display.node.alert_banner.default.yml [hunk 10]
```diff
@@ -2,35 +2,34 @@ langcode: en
...
 mode: default
 content:
   field_alert_display_date:
-    type: datetime_default
-    weight: 3
-    region: content
+    weight: 4
     label: above
     settings:
+      format_type: medium
```
Comment (sonnykt) [inline config/install/core.entity_view_display.node.alert_banner.default.yml]:
```suggestion
      format_type: alert_banner_date
```
@govindmaloo the Alert Display Date must be displayed with the date format provided by the module, originally `govcms_alert_date`.

File: config/install/field.field.node.alert_banner.field_alert_permanent.yml [hunk 11]
```diff
@@ -3,15 +3,15 @@ status: true
...
-    - node.type.govcms_alert
-id: node.govcms_alert.field_alert_permanent
+    - node.type.alert_banner
+id: node.alert_banner.field_alert_permanent
 field_name: field_alert_permanent
 entity_type: node
-bundle: govcms_alert
+bundle: alert_banner
 label: Permanent
 description: 'Permanent alerts always display even if their alert date does not fall within 7 days. Alerts with future date will not show.'
```
Comment (sonnykt) [inline config/install/field.field.node.alert_banner.field_alert_permanent.yml]:
```suggestion
description: 'Permanent alerts cannot be dismissed by visitors.'
```
@govindmaloo  please see the Solution Direction.

File: config/install/views.view.alert_banners.yml [hunk 12]
```diff
@@ -21,10 +21,8 @@ dependencies:
     - taxonomy
     - text
     - user
-_core:
-  default_config_hash: DIlesCAGPh27zNXU0V5aIneag5WMJeRnPS20IOUAj4g
-id: govcms_alerts
-label: 'GovCMS Alerts'
+id: alert_banners
+label: Alerts
```
Comment (sonnykt) [inline config/install/views.view.alert_banners.yml]:
```suggestion
label: Alert Banners
```

File: config/install/views.view.alert_banners.yml [hunk 13]
```diff
@@ -398,7 +254,7 @@ display:
     display_options:
       display_extenders:
         metatag_display_extender: {  }
-      path: govcms-alerts
+      path: alerts
```
Comment (sonnykt) [inline config/install/views.view.alert_banners.yml]:
```suggestion
      path: alert-banners
```

File: config/install/views.view.alert_banners.yml [hunk 14]
```diff
@@ -380,7 +237,6 @@ display:
         operator: AND
         groups:
           1: AND
-          2: OR
       css_class: block-digitalwa-alerts
```
Comment (sonnykt) [inline config/install/views.view.alert_banners.yml]:
```suggestion
      css_class: block-alert-banners
```

File: config/install/views.view.alert_banners.yml [hunk 15]
```diff
@@ -920,7 +776,7 @@ display:
           type: datetime_default
           settings:
             timezone_override: ''
-            format_type: govcms_alert_date
+            format_type: alert_date
```
Comment (sonnykt) [inline config/install/views.view.alert_banners.yml]:
```suggestion
            format_type: alert_banner_date
```

File: css/alert_banner.css [hunk 16]
```diff
@@ -1,46 +1,47 @@
-.govcms-alerts {
+.alerts-banners {
   position: relative;
 }
-.path-admin .govcms-alerts.layout-container {
+.path-admin .alerts-banners.layout-container {
   margin: 0;
 }
-.govcms-alerts article.node--type-alert {
+.alert-banners article.node--type-alert {
```
Comment (sonnykt) [inline css/alert_banner.css]: @govindmaloo all references of `node--type-alert` need to be changed to `node--type-alert-banner`.

File: css/alert_banner.css [hunk 17]
```diff
@@ -158,13 +161,13 @@
   -ms-transform: rotate(45deg);
   -o-transform: rotate(45deg);
   filter: progid:DXImageTransform.Microsoft.BasicImage(rotation=0.5);
-  line-height: 1;
+  line-height: 0;
 }
-.govcms-alerts article.node--type-alert .govcms-alert-message time {
+.alert-banners article.node--type-alert .alert-message time {
```
Comment (sonnykt) [inline css/alert_banner.css]:
```suggestion
.alert-banners article.node--type-alert-banner .alert-banner-message time {
```
@govindmaloo `govcms-alert` must be changed to `alert-banner`.

File: js/alert_banner.js [hunk 18]
```diff
@@ -42,10 +42,10 @@
     return true;
   }
 
-  Drupal.behaviors.govcmsAlertsRestBlock = {
+  Drupal.behaviors.AlertsRestBlock = {
     attach: function (context, settings) {
       // Set alert text color.
-      $('.govcms-alerts article.node--type-alert', context).each(function (index, element) {
+      $('.alert-banners article.node--type-alert', context).each(function (index, element) {
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
      $('.alert-banners article.node--type-alert-banner', context).each(function (index, element) {
```

File: js/alert_banner.js [hunk 19]
```diff
@@ -61,17 +61,17 @@
       });
 
       // Process the Close button of each alert.
-      $('.govcms-alerts article.node--type-alert button.govcms-alert-close', context).click(function (event) {
+      $('.alert-banners article.node--type-alert button.alert-close', context).click(function (event) {
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
      $('.alert-banners article.node--type-alert-banner button.alert-banner-close', context).click(function (event) {
```

File: js/alert_banner.js [hunk 20]
```diff
@@ -61,17 +61,17 @@
...
         var alert_id = $(event.target).attr('data-alert-id');
         $.cookie('hide_alert_id_' + alert_id, true);
         $('article.node--type-alert[data-alert-id="' + alert_id + '"]').remove();
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
        $('article.node--type-alert-banner[data-alert-id="' + alert_id + '"]').remove();
```

File: js/alert_banner.js [hunk 21]
```diff
@@ -92,7 +92,7 @@
               }
 
               // Build the alert.
-              var $alert = $('<article role="article" data-alert-id="' + alert_item.alert_id + '" class="node node--type-alert"><div class="container node__content"></div></article>');
+              var $alert = $('<article role="article" data-alert-id="' + alert_item.alert_id + '" class="node node--type-alert"><div class="layout-container node__content"></div></article>');
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
              var $alert = $('<article role="article" data-alert-id="' + alert_item.alert_id + '" class="node node--type-alert-banner"><div class="layout-container node__content"></div></article>');
```

File: js/alert_banner.js [hunk 22]
```diff
@@ -124,23 +124,23 @@
                 if ((typeof alert_item.display_date !== 'undefined') && (alert_item.display_date !== "")) {
                   alert_message = alert_item.display_date + alert_message;
                 }
-                $('<div class="field clearfix govcms-alert-message">' + alert_message + '</div>')
+                $('<div class="field clearfix alert-message">' + alert_message + '</div>')
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
                $('<div class="field clearfix alert-banner-message">' + alert_message + '</div>')
```

File: js/alert_banner.js [hunk 23]
```diff
@@ -124,23 +124,23 @@
...
                   .appendTo($alert.find('.node__content'));
               }
 
               // Attaches the link.
               if (typeof alert_item.link !== 'undefined' && alert_item.link !== false && alert_item.link !== "") {
-                $('<div class="field clearfix govcms-alert-link au-cta-link">' + alert_item.link + '</div>')
+                $('<div class="field clearfix alert-link">' + alert_item.link + '</div>')
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
                $('<div class="field clearfix alert-banner-link">' + alert_item.link + '</div>')
```

File: js/alert_banner.js [hunk 24]
```diff
@@ -124,23 +124,23 @@
...
                   .appendTo($alert.find('.node__content'));
               }
               else {
-                $alert.addClass('alert-cta-link--none');
+                $alert.addClass('alert-link--none');
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
                $alert.addClass('alert-banner-link--none');
```

File: js/alert_banner.js [hunk 25]
```diff
@@ -124,23 +124,23 @@
...
               }
 
               // Generates the Close button.
               if ((typeof alert_item.permanent !== 'undefined') && (alert_item.permanent !== '1')) {
                 var label_dismiss = Drupal.t('Dismiss alert');
-                $('<button class="govcms-alert-close" data-alert-id="' + alert_item.alert_id + '" aria-label="' + label_dismiss + '"><span>' + label_dismiss + '</span></button>')
+                $('<button class="alert-close" data-alert-id="' + alert_item.alert_id + '" aria-label="' + label_dismiss + '"><span>' + label_dismiss + '</span></button>')
```
Comment (sonnykt) [inline js/alert_banner.js]:
```suggestion
                $('<button class="alert-banner-close" data-alert-id="' + alert_item.alert_id + '" aria-label="' + label_dismiss + '"><span>' + label_dismiss + '</span></button>')
```

**Review by sonnykt — CHANGES_REQUESTED** [review]

**Comment by govindmaloo** [comment]
@sonnykt Updated js and css as per feedback.

Seven theme
Screenshot 2020-05-14 at 7 49 37 PM

Bartik theme
Screenshot 2020-05-14 at 7 49 46 PM

File: js/alert_banner.js (hunk 1 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: js/alert_banner.js (hunk 2 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: js/alert_banner.js (hunk 3 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: js/alert_banner.js (hunk 4 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: js/alert_banner.js (hunk 21 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: js/alert_banner.js (hunk 22 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: js/alert_banner.js (hunk 23 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt updated

File: js/alert_banner.js (hunk 24 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: js/alert_banner.js (hunk 25 above)
Comment (govindmaloo) [reply js/alert_banner.js]: @sonnykt Updated

File: alert_banner.module (hunk 5 above)
Comment (govindmaloo) [reply alert_banner.module]: @sonnykt Updated

File: alert_banner.module (hunk 6 above)
Comment (govindmaloo) [reply alert_banner.module]: @sonnykt Updated

File: alert_banner.routing.yml (hunk 7 above)
Comment (govindmaloo) [reply alert_banner.routing.yml]: @sonnykt Updated

File: README.md (hunk 8 above)
Comment (govindmaloo) [reply README.md]: @sonnykt Updated

File: config/install/core.date_format.alert_date.yml (hunk 9 above)
Comment (govindmaloo) [reply config/install/core.date_format.alert_date.yml]: @sonnykt Updated

File: config/install/core.entity_view_display.node.alert_banner.default.yml (hunk 10 above)
Comment (govindmaloo) [reply config/install/core.entity_view_display.node.alert_banner.default.yml]: @sonnykt Updated

File: config/install/field.field.node.alert_banner.field_alert_permanent.yml (hunk 11 above)
Comment (govindmaloo) [reply config/install/field.field.node.alert_banner.field_alert_permanent.yml]: @sonnykt Updated.

File: config/install/views.view.alert_banners.yml (hunk 12 above)
Comment (govindmaloo) [reply config/install/views.view.alert_banners.yml]: @sonnykt  Updated.

File: config/install/views.view.alert_banners.yml (hunk 14 above)
Comment (govindmaloo) [reply config/install/views.view.alert_banners.yml]: @sonnykt Updated

File: config/install/views.view.alert_banners.yml (hunk 15 above)
Comment (govindmaloo) [reply config/install/views.view.alert_banners.yml]: @sonnykt updated.

File: css/alert_banner.css (hunk 16 above)
Comment (govindmaloo) [reply css/alert_banner.css]: @sonnykt Updated all instance

File: css/alert_banner.css (hunk 17 above)
Comment (govindmaloo) [reply css/alert_banner.css]: @sonnykt Updated

**Review by sonnykt — APPROVED** [review]

File: alert_banner.install [hunk 26]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * @file
+ * Alert installation.
+ */
+
+use Drupal\taxonomy\Entity\Term;
+use Drupal\user\Entity\Role;
```
Comment (sonnykt) [inline alert_banner.install]: @govindmaloo it's failing CI
```
FILE: ...leci/project/build/web/modules/alert_banner/alert_banner.install
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 9 | WARNING | [x] Unused use statement
   |         |     (Drupal.Classes.UnusedUseStatement.UnusedUse)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Review by sonnykt — CHANGES_REQUESTED** [review]

File: alert_banner.install (hunk 26 above)
Comment (govindmaloo) [reply alert_banner.install]: @sonnykt  fixed

**Comment by govindmaloo** [comment]
@sonnykt All test pass.

**Review by sonnykt — APPROVED** [review]

**Pull request merged**